### PR TITLE
Add support for padding around an array

### PR DESCRIPTION
**Pull request by ZacCranko**
To address #9. I should add that the order of the arguments here has been tweaked slightly (compared with #9) to be more consistent with the existing methods.

### CODE REVIEW

File: src/PaddedViews.jl [hunk 1]
```diff
@@ -2,16 +2,20 @@ __precompile__(true)
...
+    datapadded = PaddedView(fillvalue, data, padded_indices)
+    datapadded = PaddedView(fillvalue, data, padded_indices, data_indices)
 
 Create a padded version of the array `data`, where any elements within
-the span of `indices` not assigned in `data` will have value
-`fillvalue`.
+the span of `padded_indices` not assigned in `data` will have value
+`fillvalue`. If a second set of indices `data_indices` is not supplied
+it is assumed the array `data` spans the indices from the first up until
+`size(data)`, otherwise `data` spans the specified `data_indices`.
```
Comment (timholy) [inline src/PaddedViews.jl]: This doesn't explain what the version with an `NTuple{N,Int}` for `data_indices` does. (That's described implicitly for the third argument with the `sz` line, though one could argue that should be more explicit too.)

**Review by timholy — COMMENTED** [review]
Just one minor suggested change for the docstring, and this is good to go.

**Comment by ZacCranko** [comment]
Cheers, done! Do you know what's going with the ambiguities test failing?

**Comment by timholy** [comment]
Best way to do that is to run the tests in a REPL and inspect the contents of `setdiff(detect_ambiguities(PaddedViews, Base, Core), ambs)`.

**Comment by Evizero** [comment]
From looking at the code diff it doesn't look like this PR is the direct cause of an ambiguity. Instead I suspect the introduction of the OffsetArrays package just needs to be taken into account for `ambs` in the test file.

@ZacCranko are you still interested in looking into this further? Alternatively we could merge this as as and I can do the remaining little tweaks.

**Comment by ZacCranko** [comment]
Apologies, I forgot about this. Yes please go ahead and merge as is if you're good with that @Evizero. Thanks!

**Comment by kleinschmidt** [comment]
Is there anything holding this up? This feature came up on slack recently.

**Comment by Evizero** [comment]
I don't have any permissions here, but if this does get merged I can look into fixing the ambiguities issues later this week (after wednesday)

**Comment by timholy** [comment]
See #12

**Comment by timholy** [comment]
Thanks @ZacCranko!

**Comment by ZacCranko** [comment]
My pleasure. Cheers @timholy and @Evizero!

**Pull request merged**